tfjs-backend-webgpu/src/kernels/reduce_webgpu.ts

```diff
@@ -41,32 +41,25 @@ export class ReduceProgram implements WebGPUProgram {
         backend_util.computeOutAndReduceShapes(this.inputShape, [1]);
     this.outputShape = outputShape.length === 0 ? [1] : outputShape;
 
-    this.reductionFactor = 2;
-    // Note that the maximum of workgroup X dimension is 256.
-    const xMaxThreads = 256;
-    const xThreads = Math.min(
-        Math.ceil(reduceInfo.inSize / this.reductionFactor), xMaxThreads);
-
-    this.workGroupSize = [xThreads, 1, 1];
-    this.dispatchLayout = {x: [], y: this.outputShape.map((d, i) => i)};
-    this.dispatch = computeDispatch(
-        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    // A work group only outputs a data, so we transfer [1, 1, 1] to compute
+    // dispatch size.
+    this.dispatch =
+        computeDispatch(this.dispatchLayout, this.outputShape, [1, 1, 1]);
 
     this.reduceType = reduceType;
     this.shaderKey = `reduce_${reduceType}_${outputDtype}`;
```

Comment:
nit: Can you also remove `outputDtype` from shaderKey? It seems that `outputDtype` is useless.